Lesson4_wordbreak.cpp

```diff
@@ -0,0 +1,136 @@
+/*Word Break
+Given an input string and a dictionary of words, find out if the input string can be segmented into a space-separated sequence of dictionary words.
+For example, consider the following dictionary: { pear, salmon, foot, prints, footprints, leave, you, sun, girl, enjoy },
+
+Examples:
+Given the string “youenjoy”, 
+Output: True (The string can be segmented as “you enjoy”)
+
+Input: “youleavefootprints”,
+Output: True (The string can be segmented as “you leave footprints” or “you leave foot prints”)
+
+Input:salmonenjoyapples
+Output: False
+*/
+#include<iostream>
+#include<cstring>
+#include<vector>
+using namespace std;
+bool findword(string word,vector<string> dict)
+{
+	for(int i=0;i<dict.size();++i)
+	{
+		if(!(dict[i].compare(word)))
+			return true;
+	}
+	return false;
+	
+}
+bool wordbreak(string str,vector<string> dict)
+{
+	int strlen,dictlen;
+	strlen=str.size();
+	dictlen=dict.size();
```

Comment:
it's better to declare your variables whenever you initialize them, so here it is clearer to write
int strlen = str.size();
int dictlen = dict.size();

Reply:
I will keep that in mind . Thanks!